shared/protocol/src/main/java/io/pravega/shared/protocol/netty/AppendDecoder.java

```diff
@@ -130,30 +125,34 @@ public Request processCommand(WireCommand command) throws Exception {
     }
 
     private ByteBuf getAppendDataBuf(WireCommands.AppendBlockEnd blockEnd, int sizeOfWholeEventsInBlock) throws IOException {
-        ByteBuf appendDataBuf = currentBlock.getData().slice(0, sizeOfWholeEventsInBlock);
-        int remaining = currentBlock.getData().readableBytes() - sizeOfWholeEventsInBlock;
-        if (remaining > 0) {
-            ByteBuf dataRemainingInBlock = currentBlock.getData().slice(sizeOfWholeEventsInBlock, remaining);
-            WireCommand cmd = CommandDecoder.parseCommand(dataRemainingInBlock);
-            if (!(cmd.getType() == WireCommandType.PARTIAL_EVENT || cmd.getType() == WireCommandType.PADDING)) {
-                throw new InvalidMessageException("Found " + cmd.getType()
-                        + " at end of append block but was expecting a partialEvent or padding.");
-            }
-            if (cmd.getType() == WireCommandType.PADDING && blockEnd.getData().readableBytes() != 0) {
-                throw new InvalidMessageException("Unexpected data in BlockEnd");
-            }
-            if (cmd.getType() == WireCommandType.PARTIAL_EVENT) {
-                // Work around a bug in netty:
-                // See https://github.com/netty/netty/issues/5597
-                if (appendDataBuf.readableBytes() == 0) {
-                    appendDataBuf.release();
-                    return wrappedBuffer(((WireCommands.PartialEvent) cmd).getData(), blockEnd.getData());
-                } else {
-                    return wrappedBuffer(appendDataBuf, ((WireCommands.PartialEvent) cmd).getData(), blockEnd.getData());
+        if (currentBlock != null) {
```

Comment:
Formatting suggestion:
Replace this `currentBlock != null` with `currentBlock == null` and bail out early in that case (your current `else` block). Then, without writing an explicit `else` block, keep the original code. That will save you from looking like you changed this entire method.

Reply:
yes .. its changed now.